src/jailer/src/env.rs

```diff
@@ -481,4 +488,80 @@ mod tests {
             libc::close(fd2);
         }
     }
+
+    fn skip_not_found(err: std::io::Error) -> std::io::Result<()> {
+        match err.kind() {
+            std::io::ErrorKind::NotFound => Ok(()),
+            _ => Err(err),
+        }
+    }
+
+    fn get_major(dev: u64) -> u32 {
+        unsafe { libc::major(dev) }
+    }
+
+    fn get_minor(dev: u64) -> u32 {
+        unsafe { libc::minor(dev) }
+    }
+
+    #[test]
+    fn test_mknod_and_own_dev() {
+        use std::os::unix::fs::{FileTypeExt, MetadataExt, PermissionsExt};
+
+        // Create environment with good arguments.
+        let node = "1";
+        let id = "bd65600d-8669-4903-8a14-af88203add38";
+        let exec_file = "/proc/cpuinfo";
+        let uid = "1001";
+        let gid = "1002";
+        let chroot_base = "/";
+        let netns = Some("zzzns");
+
+        let good_arg_vals = ArgVals {
```

Comment:
nit: I would create the ArgVals like this in order to save some lines:
```
let arg_vals = ArgsVals {
    node: "1",
    id: "bd65600d-8669-4903-8a14-af88203add38",
    ....
}
```

Reply:
Thanks, this is a lot better! See c38904dfb7e04a2a8e229394a0145c409edfc614.